pkg/expr/service.go

```diff
@@ -34,9 +36,13 @@ const DatasourceID = -100
 // should be used instead and should be set to "__expr__".
 const OldDatasourceUID = "-100"
 
-// IsDataSource checks if the uid points to an expression query
-func IsDataSource(uid string) bool {
```

Comment:
Can we keep this and call it inside of QueryKindByDatasourceUID (also rename to become NodeTypeFromDatasourceUID). The reason is to is that It seems confusing when looking at the query package to have to being looking at the NodeType comparison without knowing a bit more about the expr package then seems is needed.